lib/common/client/middleware/logging.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Middleware
+      class Logging < Faraday::Middleware
+        DEFAULT_TYPE_KEY = 'undefined'
+
+        def initialize(app, type_key)
+          super(app)
+          @type_key = type_key || DEFAULT_TYPE_KEY
+        end
+
+        def call(env)
+          request_body = Base64.encode64(env.body)
```

Comment:
VCR records ASCII-8BIT, which doesn't dump to JSON well in this case. Nice thing that this was caught immediately, since the middleware could technically be used to record binary contents that also wouldn't serialize. We'll just need to write a processor to analyze these records when we have them (we'd need that anyway)